code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm

```diff
@@ -865,3 +865,61 @@
 		log_game("DYNAMIC: Not enough players volunteered for the ruleset [name] - [candidates.len] out of [required_candidates].")
 		return FALSE
 	return TRUE
+
+//////////////////////////////////////////////
+//                                          //
+//           NINJA      (GHOST)             //
+//                                          //
+//////////////////////////////////////////////
+
+/datum/dynamic_ruleset/midround/from_ghosts/ninja
+	name = "Space Ninja"
+	midround_ruleset_style = MIDROUND_RULESET_STYLE_HEAVY
+	role_preference = /datum/role_preference/midround_ghost/ninja
+	required_type = /mob/dead/observer
+	antag_datum = /datum/antagonist/ninja
+	enemy_roles = list(JOB_NAME_SECURITYOFFICER, JOB_NAME_DETECTIVE, JOB_NAME_WARDEN, JOB_NAME_HEADOFSECURITY, JOB_NAME_CAPTAIN)
+	required_enemies = list(2,2,2,2,2,2,2,2,2,2)
+	required_candidates = 1
+	weight = 3
+	cost = 9
+	minimum_players = 20
+	repeatable = TRUE
+	var/spawn_loc
+
+/datum/dynamic_ruleset/midround/from_ghosts/ninja/ready(forced)
+	if (!..())
+		return FALSE
+	//selecting a spawn_loc
+	var/list/spawn_locs = list()
+	for(var/obj/effect/landmark/carpspawn/L in GLOB.landmarks_list)
+		if(isturf(L.loc))
+			spawn_locs += L.loc
+	if(!length(spawn_locs))
+		log_game("DYNAMIC: [ruletype] ruleset [name] ready() failed due to no valid spawn locations (#1).")
+		return FALSE
+	spawn_loc = pick(spawn_locs)
+	if(!spawn_loc)
+		log_game("DYNAMIC: [ruletype] ruleset [name] ready() failed due to no valid spawn locations (#2).")
+		return FALSE
+	return TRUE
+
+/datum/dynamic_ruleset/midround/from_ghosts/ninja/generate_ruleset_body(mob/applicant)
+	//spawn the ninja and assign the candidate
+	var/mob/living/carbon/human/Ninja = create_space_ninja(spawn_loc)
+
+	//Prepare ninja player mind
+	// Dynamic's finish_setup proc will handle application of antagonist datums
+	var/datum/mind/Mind = new /datum/mind(applicant.key)
+	Mind.active = TRUE
+	Mind.transfer_to(Ninja)
+
+	message_admins("[ADMIN_LOOKUPFLW(Ninja)] has been made into a ninja by the midround ruleset")
+	log_game("[key_name(Ninja)] was spawned as a ninja by the midround ruleset.")
+
+	return Ninja
+
+/datum/dynamic_ruleset/midround/from_ghosts/ninja/finish_setup(mob/new_character, index)
+	. = ..()
+	// Set their job in addition to their antag role to be a space ninja for logging purposes
+	new_character.mind.assigned_role = ROLE_NINJA
```

Comment:
This seems unnecessary

Reply:
Also setting the job to something invalid causes runtimes

Reply:
A few logging things use assigned_role and other midround antagonists like nukie have their assigned role set to their antagonist role.

Reply:
> Also setting the job to something invalid causes runtimes

This is currently a standard approach for midround antagonists, obsession brain trauma is the only place this can runtime but I would probably label that as an obsession issue as it misuses the GetJob proc.
![image](https://github.com/BeeStation/BeeStation-Hornet/assets/26465327/ecf6a286-93d2-407c-8f20-1976c64c3c38)
![image](https://github.com/BeeStation/BeeStation-Hornet/assets/26465327/7a04e7eb-72ba-4917-a0ae-162fa103f107)
I will label this as an obsession issue since by the way this is coded it seems desisgned specifically to runtime. There are no other cases in the code I can find where assigned_role is used in a way that could cause runtimes. All other cases can handle, or specifically handle, antagonist roles being allows in assigned_role for cases where the player was not spawned on the station.